server/libs/FileUploader.php

```diff
@@ -56,6 +56,27 @@ public function upload($fileKey) {
         }
     }
 
+    public function uploadCSVFile($fileKey) {
```

Comment:
If we do it this way, we will have some code duplication (between this function, and function `upload`). This is not ideal, since later if we need to update one, we will always have to keep in mind that there is another, quite similar, function out there. It might not seem so important now but it can become a pain.

So how do we deal with this? We factor out the "common denominator".

For example, you can see both of these functions do exactly the same thing, _except_, for the part on the validations, the normal `upload` checks many possible file types, while the one for CSV only checks of CSV and plain text.

Thus, we can create a function `uploadFile`, that must be **private** (we only want to use it internally). This function will receive the `$fileKey`, but it will also receive an array `$fileTypes`. Then it will do all the same things, but passing this array to the `\Upload\Validation\Mimetype`.

Then, how would the implementation of these `upload` and `uploadCSVFIle` be in terms of this new function `uploadFile`? We can do them like this:

```php
public function uploadCSVFile($fileKey) {
    return $this->uploadFile($fileKey, [
        'text/csv',
        'text/plain'
    ]);
}
```

And the other one would be similar, just that passing the longer array of file types instead.

But there it is, now if there's a change in some of the functionality of the function, then it will be done **in a single place**.